### DISCUSSION TITLE
cascading form - help to understand

### DISCUSSION THREAD

**Discussion by Pfurr** [react-hook-form/react-hook-form#5629] [Q&A]
i need to filtering data by cascading dropdown and send to server user selection.

i think it's possible to do with react hook form. Can someone help me? Thank you cc -> @bluebill1049 

I add my code here: https://codesandbox.io/s/react-hook-form-cascading-select-dev-vmthy?file=/src/Test.js

**Comment by bluebill1049** [comment]
> i need to filtering data by cascading dropdown and send to server user selection.

what do you mean? you want to select something and return it from the server and then update UI? probably don't need hook form.

**Reply by Pfurr** [reply]
i'm working with a form and I have to filter a large object (see code) and then send the user selections to the server by form (that has others input)

**Reply by Pfurr** [reply]
Ps. And is there an example of Cascading Selects with RHFORM v.7? @bluebill1049

**Comment by thanh-nguyen-95** [comment]
You have a working version in the CodeSandbox. All that's left is to fetch real data and managing the values when user selected. I don't see any problem related to `react-hook-form` here.

**Reply by Pfurr** [reply]
> You have a working version in the CodeSandbox. All that's left is to fetch real data and managing the values when user selected. I don't see any problem related to `react-hook-form` here.

of course, it says need Help on it! if you do not want to help is OK, but I don't understand your comment that's no help

**Reply by thanh-nguyen-95** [reply]
My apology if my comment upset you in any way.

My point is that your question was not clear enough. You provided a working version of cascading selects using `react-hook-form` (be it without the fetching data part) and asked for help. What exactly do you need help with? For example:

- You copy the code somewhere else and don't understand what it does
- You know how to fetch the data but don't know how to set the data back to `react-hook-form`
- You did everything but the result is not what you expected

Clear questions get clear answers. Please specify what you need so people can help you more effectively.

Again, I didn't try to be mean. I just need to know how to help you. I wouldn't comment if I didn't want to help.

**Reply by Pfurr** [reply]
> My apology if my comment upset you in any way.
> 
> My point is that your question was not clear enough. You provided a working version of cascading selects using `react-hook-form` (be it without the fetching data part) and asked for help. What exactly do you need help with? For example:
> 
> * You copy the code somewhere else and don't understand what it does
> * You know how to fetch the data but don't know how to set the data back to `react-hook-form`
> * You did everything but the result is not what you expected
> 
> Clear questions get clear answers. Please specify what you need so people can help you more effectively.
> 
> Again, I didn't try to be mean. I just need to know how to help you. I wouldn't comment if I didn't want to help.

No problem, now i understand your point of view :)  Anyway:

---- My code not working -> look -> FORM TO DEVELOPING... -> What I wanted to do is remove the duplicates of the JSON, and filter all the JSON in this way: 
1 - filtering list by types of shops;  
2 - filtering list types store by country, f.e. type of shop in each country;  
3 - filtering list country store by province ( if it exists),
4 -  filtering list province store by city;  
5 -  filtering list city store by business_name;  
6 -  filtering list business_name store by owned store or not owned; 
7 -  filtering list owned store  store by store_code store ...

But it doesn't work well, the selectors seem to be not correlating and don't filtering array of obj  matching the correct criteria


---- Then  i was looking an example of cascading form with rhf ,  because I work a little bit of an legacy version, and when I updated RHF, it return me a series of issues on ref and userForm for example...

**Reply by bluebill1049** [reply]
@thanh-nguyen-95 thank you for your support and kindness. 🙏

**Reply by thanh-nguyen-95** [reply]
@pherm Hi there, sorry for the late response. I've been busy over the week.

Here's an example of how to create cascading selects with `react-hook-form` v7: https://codesandbox.io/s/react-hook-form-cascading-select-dev-forked-r9xy9?file=/src/Test.js

I only make one "releationship" between country and types of shop, please do the rest yourself. In general, you need to:

- Use `watch` to get filter values. It will get updated automatically everytime user changes the value (if it's registered)
- Use `useMemo` to filter the list based on filter values (country, city, ...). Make sure you declare enough dependecies for it to work correctly.

Hope this helps. If you have any further question, feel free to ask. Good luck!

Ps: I upgrade all dependencies (libraries) to their latest versions in my CodeSandbox. Just want to leave it here in case you didn't notice
